spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/repository/query/PartTreeFirestoreQuery.java

```diff
@@ -86,6 +91,19 @@ public Object execute(Object[] parameters) {
 
 		StructuredQuery.Builder builder = createBuilderWithFilter(parameters);
 
+
+		// Handle Pageable parameters.
+		if (!getQueryMethod().getParameters().isEmpty()) {
+			ParameterAccessor paramAccessor = new ParametersParameterAccessor(getQueryMethod().getParameters(),
+					parameters);
+			Pageable pageable = paramAccessor.getPageable();
+			if (pageable != null && pageable.isPaged()) {
+				builder.setOffset((int) Math.min(Integer.MAX_VALUE, pageable.getOffset()));
```

Comment:
This is interesting - `setOffset` accepts an `int`, but `setLimit` needs a `Int32Value`? :)

Reply:
Lol beats me... Yeah that is what RunQueryRequest accepts :open_mouth: 

See the object definition: https://cloud.google.com/firestore/docs/reference/rpc/google.firestore.v1#structuredquery